dmaws/hosted_graphite/create_alerts.py

```diff
@@ -0,0 +1,134 @@
+        "alert_criteria": {
+            "type": "above",
+            "above_value": 0
+        },
+        "notification_channels": ["Notify DM 2ndline"],
+        "notification_type": ["every", 60],
+        "info": """429s responses being returned from the production router. Check CloudWatch for the IP address to make
+sure this is a crawler rather than a legitimate request"""
+    },
+    {
+        "name": "Production Router slow requests (10+ seconds)",
+        "metric": "cloudwatch.request_time_buckets.production.router.request_time_bucket_9.sum",
+        "alert_criteria": {
+            "type": "above",
+            "above_value": 5
+        },
+        "notification_channels": ["Notify DM 2ndline"],
+        "notification_type": ["every", 60],
+        "info": "5+ requests taking 10+ seconds from the production router in the last minute"
+    },
+    {
+        "name": "Production Router slow requests (5-10 seconds)",
+        "metric": "cloudwatch.request_time_buckets.production.router.request_time_bucket_8.sum",
+        "alert_criteria": {
+            "type": "above",
+            "above_value": 5
+        },
+        "notification_channels": ["Notify DM 2ndline"],
+        "notification_type": ["every", 60],
+        "info": "5+ requests taking 5-10 seconds from the production router in the last minute"
+    },
+]
+
+
+def get_missing_logs_alert_json(environment, app):
+    """
+    For a given environment and application, return the JSON required to set up an alert that will trigger if either
+    no nginx log event metrics or no application log event metrics are received for 15 minutes.
+
+    We use an OR statement for either nginx or application logs so we can save on the number of alerts we are using. A
+    developer seeing this alert will need to manually diagnose if it is just one or both of the log types that have
+    stopped. Note, the router app only has nginx logs so we do not need to do this for the router.
+
+    These alerts are not editable through the Hosted Graphite GUI as referenced in the API documentation at time of
+    writing:
+
+    `Our UI doesn’t fully support composites at the moment. As a result, composite alerts cannot be edited via the UI -
+    it needs to be done via the API. The alert overview page (when you click the eye button on an alert) will only
+    display one metric for the alert instead of all the metrics associated. However the alert notifications are working
+    and will display the graph of the last metric that breached the alert threshold. So for example, if the alert is
+    a && b, and a breaches the threshold, then a few minutes later b breaches it’s threshold, the alert notification
+    will show the metric graph for b`
+
+    Time period for alerting missing logs is set to 15 minutes. The smoke tests run every 5 minutes, this allows
+    some wiggle room to avoid false positives arising from a delay in shipping the metrics from Cloudwatch.
```

Comment:
It feels like the part about avoiding false positives should be linked to the sentence about the time period being 15 minutes rather than linked with the 5 minute smoke test sentence